src/main/java/org/cardanofoundation/cip8/CIP8Parser.java

```diff
@@ -28,25 +28,65 @@
 import static org.cardanofoundation.ext.cbor.MoreCbor.serialize;
 import static org.cardanofoundation.ext.cose.COSEKey.deserialize;
 
+/**
+ * The {@code CIP8Parser} class is used to verify and parse a message that uses the CIP 8
+ * specification of the Cardano protocol.
+ * For further information on CIP 8, see <A href="https://cips.cardano.org/cips/cip8/">
+ * https://cips.cardano.org/cips/cip8/</A>.
+ */
 public final class CIP8Parser {
 
+    /**
+     * Static instance of {@code Logger} used for logging.
+     */
     private static final Logger logger = LoggerFactory.getLogger(CIP8Parser.class);
 
+    /**
+     * TODO
+     */
     private static final EdDSAParameterSpec ED_DSA_PARAMETER_SPEC = EdDSANamedCurveTable.getByName(ED_25519);
+
+    /**
+     * TODO
+     */
     private static final long PUBLIC_KEY_INDEX = -2;
 
+    /**
+     * Signature of the message signed using CIP 8.
+     */
     private final String signature;
+
+    /**
+     * Public key used to sign the message using CIP 8.
+     */
     private final Optional<String> publicKey;
 
+    /**
+     * Creates an {@code CIP8Parser}.
+     *
+     * @param signature a string that contains the signature of the message signed using CIP 8.
+     */
     @ParametersAreNonnullByDefault
     public CIP8Parser(String signature) {
-       this(signature, Optional.empty());
+        this(signature, Optional.empty());
     }
 
+    /**
+     * Creates an {@code CIP8Parser}.
+     *
+     * @param signature a string that contains the signature of the message signed using CIP 8.
+     * @param publicKey a string that contains the public key used to sign the message using CIP 8.
```

Comment:
I think it is worth to specify that this constructor should be used when it is anticipated that the client will not have public key encoded in signature itself.